java/ql/src/Security/CWE/CWE-079/AndroidWebViewAddJavascriptInterfaceExample.java

```diff
@@ -0,0 +1,11 @@
+class ExposedObject {
+    @JavascriptInterface
+    public String example() {
+        return "String from Java";
+    }
+}
+
+webview.getSettings().setJavaScriptEnabled(true);
+webview.addJavaScriptInterface(new ExposedObject(), "exposedObject");
+webview.loadData("", "text/html", null);
+webview.loadUrl("javascript:alert(exposedObject.example())");
```

Comment:
I would like to see a more obvious example of why this could be a security problem. Instead of the trivial "return a String" method, you could show a more interesting case, e.g. one that takes parameters from the JavaScript and runs an OS command or builds a SQL statement with them.

Reply:
Ok good idea. I'll come up with something for that